internal/controller/promotions/promotions.go

```diff
@@ -665,3 +662,20 @@ func parseCreateActorAnnotation(promo *kargoapi.Promotion) string {
 	}
 	return creator
 }
+
+func calculateRequeueInterval(p *kargoapi.Promotion) time.Duration {
+	defaultRequeueInterval := 5 * time.Minute
+	step := p.Spec.Steps[p.Status.CurrentStep]
+	runner := promotion.GetStepRunner(step.Uses)
+	timeout := ptr.To(time.Duration(0))
+	if retryCfg, isRetryable := runner.(pkgPromotion.RetryableStepRunner); isRetryable {
```

Comment:
Absolute nit: now that `var()` is gone, this makes it slightly more pleasant to my eyes to read.

```suggestion
	defaultRequeueInterval := 5 * time.Minute
	step := p.Spec.Steps[p.Status.CurrentStep]
	runner := promotion.GetStepRunner(step.Uses)
	timeout := ptr.To(time.Duration(0))

	if retryCfg, isRetryable := runner.(pkgPromotion.RetryableStepRunner); isRetryable {
```